src/assets.rs

```diff
@@ -204,17 +195,39 @@ impl HighlightingAssets {
                         self.syntax_set.find_syntax_by_name(syntax_name)
                     }
                     Some(MappingTarget::MapToUnknown) => line_syntax,
-                    None => ext_syntax.or(line_syntax),
+                    None => self.get_extension_syntax(file_name).or(line_syntax),
                 }
             }
-            (None, InputFile::StdIn) => String::from_utf8(reader.first_line.clone())
+            (None, InputFile::StdIn(None)) => String::from_utf8(reader.first_line.clone())
                 .ok()
                 .and_then(|l| self.syntax_set.find_syntax_by_first_line(&l)),
+            (None, InputFile::StdIn(Some(file_name))) => self
+                .get_extension_syntax(file_name.to_str().unwrap())
```

Comment:
Always try to avoid plain `.unwrap()` calls. This will panic if `.to_str()` fails.

Currently working on a fix.